internal/logic/logic.go

```diff
@@ -1,53 +1,122 @@
+func NewTransactionManagementServiceLogic(ds datasource.DataSourceI, as string) TransactionManagementServiceLogicIer {
 	return &transactionManagementServiceLogic{
-		dummyDsSvc: ds,
-    }
-}
-
-func (l transactionManagementServiceLogic) Ping(req *model.PingRequest) *respModel.Response {
-	// add business logic here
-	res, err := l.dummyDsSvc.Ping(&model.PingDs{
-    	Data: req.Data,
-    })
-    if err != nil {
-        log.Error("datasource error", err)
-    	return &respModel.Response{
-    		Status:  http.StatusInternalServerError,
-    		Message: "",
-    		Data:    nil,
-    	}
-    }
-    return &respModel.Response{
-    	Status:  http.StatusOK,
-    	Message: "Pong",
-    	Data:    res,
-    }
+		DsSvc:  ds,
+		AccSvc: as,
+	}
 }
 
 func (l transactionManagementServiceLogic) HealthCheck() bool {
 	// check all internal services are working fine
-	return l.dummyDsSvc.HealthCheck()
-}
\ No newline at end of file
+	return l.DsSvc.HealthCheck()
+}
+
+func (l transactionManagementServiceLogic) GetTransactions(id string, limit int, page int) *respModel.Response {
+	offset := (page - 1) * limit
+	transactions, count, err := l.DsSvc.Get(map[string]interface{}{"user_id": id}, limit, offset)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrGetTransaction),
+			Data:    nil,
+		}
+	}
+	totalPages := int(math.Ceil(float64(count) / float64(limit)))
+	nextPage := -1
+	if count-offset > limit {
+		nextPage = page + 1
+	}
+	resp := model.PaginatedResponse{Response: transactions, Pagination: model.Paginate{CurrentPage: page, NextPage: nextPage, TotalPage: totalPages}}
+	return &respModel.Response{
+		Status:  http.StatusOK,
+		Message: "SUCCESS",
+		Data:    resp,
+	}
+}
+
+func (l transactionManagementServiceLogic) NewTransaction(newTransaction model.NewTransaction) *respModel.Response {
+	transaction := model.Transaction{
+		UserId:        newTransaction.UserId,
+		AccountNumber: newTransaction.AccountNumber,
+		TransactionId: uuid.NewString(),
+		Amount:        newTransaction.Amount,
+		TransferTo:    newTransaction.TransferTo,
+		Status:        newTransaction.Status,
+		Type:          newTransaction.Type,
+		Comment:       newTransaction.Comment,
+	}
+	err := l.DsSvc.Insert(transaction)
+	if err != nil {
+		log.Error(err)
+		return &respModel.Response{
+			Status:  http.StatusInternalServerError,
+			Message: codes.GetErr(codes.ErrNewTransaction),
+			Data:    nil,
+		}
+	}
+	if newTransaction.Status != "approved" {
+		return &respModel.Response{
+			Status:  http.StatusCreated,
+			Message: "SUCCESS",
+			Data:    nil,
+		}
+	}
+	upTransaction := model.UpdateTransaction{AccountNumber: newTransaction.AccountNumber, Amount: newTransaction.Amount, TransactionType: newTransaction.Type}
+	by, err := json.Marshal(upTransaction)
+	if err != nil {
+		log.Error(codes.GetErr(codes.ErrNewTransaction))
```

Comment:
log the actual error.

Reply:
done

Reply:
Yet to be implemented.

Reply:
Implemented as part of https://github.com/vatsal278/TransactionManagementService/pull/2/commits/362ba1dc587191cc87bb585e3cc18573efd48e90